{{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/components/modal/modal.jinja

```diff
@@ -0,0 +1,29 @@
+{% from 'components/button/button.jinja' import button %}
+{% set default_button %}
+{{ button(button_content="Open Modal")}}
+{% endset %}
+{% macro modal(button=default_button, width_class="max-w-xl", open=False) %}
+<div x-data="{open:{{open|tojson}}}">
+    <span x-on:click="open=true">{{button}}</span>
+    <div
+        x-dialog
+        x-model="open"
+        style="display: none"
+        class="fixed inset-0 overflow-y-auto z-10"
+    >
+        <div x-dialog:overlay x-transition.opacity class="fixed inset-0 bg-black bg-opacity-50"></div> 
+        <div
+            class="relative min-h-screen flex items-center justify-center p-4"
+        >
+            <div
+                x-dialog:panel
+                x-transition
+                class="relative w-full bg-white rounded-xl shadow-lg overflow-y-auto {{width_class}}"
+            >
+                <!-- All header, body and footer content should be included within the macro block -->
```

Comment:
A tiny suggestion if I may! Would it be worth adding comments inside {# #} tags so these are visible for developers and not within the rendered HTML?